packages/truffle-compile/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -0,0 +1,205 @@
+const requireFromString = require("require-from-string");
+const debug = require("debug")("compile:compilerSupplier");
+const solcWrap = require("solc/wrapper");
+const findCacheDir = require("find-cache-dir");
+const originalRequire = require("original-require");
+const ora = require("ora");
+const request = require("request-promise");
+const fs = require("fs");
+const semver = require("semver");
+
+class LoadingStrategy {
+  constructor(options) {
+    const defaultConfig = {
+      versionsUrl: "https://solc-bin.ethereum.org/bin/list.json",
+      compilerUrlRoot: "https://solc-bin.ethereum.org/bin/",
+      dockerTagsUrl:
+        "https://registry.hub.docker.com/v2/repositories/ethereum/solc/tags/",
+      cache: true
+    };
+    this.config = Object.assign({}, defaultConfig, options);
+    this.cachePath = findCacheDir({
+      name: "truffle",
+      cwd: __dirname,
+      create: true
+    });
+  }
```

Comment:
There should be an abstract `load` method here to indicate that it's the external interface.

Something like:

```javascript

  async load (userSpecification) {
    throw new Error("Abstract method `loadingStrategy.load` is not implemented for this strategy.");
  }
```

Reply:
There should also be docstrings for every public method here, vs. the methods that are only used by other loading strategies.